In [ ]:
import pandas as pd
import numpy as np
import matplotlib as mlt
import scipy.stats as ss

In [ ]:
PATH = 'EVALMASTER.xlsx'

In [ ]:
df = pd.read_excel(PATH)
print(df.head(5))
df.describe()

   Grupo de control  Nivel glucosa basal  Nivel glucosa 60 min
0                 1                   90                   159
1                 1                   82                   151
2                 1                   80                   148
3                 1                   75                   138
4                 1                   74                   141


,Grupo de control,Nivel glucosa basal,Nivel glucosa 60 min
count,65.000000,65.000000,65.000000
mean,1.461538,86.861538,160.292308
std,0.502398,8.462928,14.231043
min,1.000000,65.000000,131.000000
25%,1.000000,80.000000,149.000000
50%,1.000000,88.000000,159.000000
75%,2.000000,93.000000,172.000000
max,2.000000,104.000000,197.000000


# EJERCICO 1
## A. Medidas de centralización

In [ ]:
# Agrupar por 'Grupo de control' y calcular medidas de centralización y dispersión para 'Nivel glucosa basal'
grouped = df.groupby('Grupo de control')['Nivel glucosa basal']
stats = grouped.agg(['count','min','max','mean','median'])
# Calcular la(s) moda(s) por grupo (puede haber más de una)
modes = grouped.apply(lambda s: s.mode().tolist())
stats['mode'] = modes
# Dispersion
stats_dispersion = grouped.agg(['std','var'])
iqr = grouped.quantile(0.75) - grouped.quantile(0.25)
stats['std'], stats['var'] = stats_dispersion['std'], stats_dispersion['var']
stats['IQR'] = iqr

sk_by_group = df.groupby('Grupo de control')['Nivel glucosa basal'] \
                .apply(lambda x: ss.skew(x.dropna(), bias=False))

print(stats)

                  count  min  max       mean  median              mode  \
Grupo de control                                                         
1                    35   65  103  84.685714    82.0  [75, 79, 82, 90]   
2                    30   77  104  89.400000    90.0              [88]   

                       std        var   IQR  
Grupo de control                             
1                 8.844312  78.221849  12.0  
2                 7.346592  53.972414   9.5  


A) La media es una claramente respresentativa, en grupo 2 casi perfecta. Las medias y medianas son muy parecidas y el coeficeinte de variacion es ~0, aunque el grupo 1 tiene una variacion ligeramente positiva.

In [ ]:
#Cálculo de simetría y curtosis para el grupo 2 (adultos)
col_group = 'Grupo de control'
col_basal = 'Nivel glucosa basal'
x = df.loc[df[col_group] == 2, col_basal].dropna()
skew = ss.skew(x, bias=False)
kurt_excess = ss.kurtosis(x, fisher=True, bias=False)
pearson_kurt = ss.kurtosis(x, fisher=False, bias=False)
print('--- Repetición: Grupo 2 (adultos) ---')
print('n =', len(x))
print(f'Skewness: {skew:.6f}')
print(f'Excess kurtosis (Fisher): {kurt_excess:.6f}')
print(f'Pearson kurtosis: {pearson_kurt:.6f}')
print(x.describe())

** Resultados para Nivel glucosa basal en adultos (grupo 2)**

- **Simetría (skewness):** -0.00747 (prácticamente 0 → distribución simétrica)
- **Curtosis (exceso):** -0.61266 (platykurtic — colas/peak más planas que la normal)
- **Curtosis (Pearson):** 2.38734
- **Media:** 89.40, **Desviación estándar:** 7.35